yt_dlp/extractor/radiko.py

```diff
@@ -0,0 +1,228 @@
+                'l': '15',
+                'lsid': '77d0678df93a1034659c14d6fc89f018',
+                'type': 'b',
+            })
+            if playlist_url in found:
+                continue
+            else:
+                found.add(playlist_url)
+
+            time_to_skip = None if is_onair else cursor - ft
+
+            subformats = self._extract_m3u8_formats(
+                playlist_url, video_id, ext='m4a', entry_protocol='m3u8',
+                live=True, fatal=False, m3u8_id=None,
+                headers={
+                    'X-Radiko-AreaId': area_id,
+                    'X-Radiko-AuthToken': auth_token,
+                })
+            for sf in subformats:
+                domain = sf['format_id'] = compat_urllib_parse.urlparse(sf['url']).netloc
+                if re.match(r'^[cf]-radiko\.smartstream\.ne\.jp$', domain):
+                    sf['preference'] = 100 if is_onair else -100  # current radio stream
+                if not is_onair and url_attrib['timefree'] == '1' and time_to_skip:
+                    # sf['format_note'] = 'timefree'
+                    sf['_ffmpeg_args'] = ['-ss', time_to_skip]
+            formats.extend(subformats)
+
+        self._sort_formats(formats)
+        return formats
+
+
+class RadikoIE(RadikoBaseIE):
+    _VALID_URL = r'https?://(?:www\.)?radiko\.jp/#!/ts/(?P<station>[A-Z0-9-]+)/(?P<id>\d+)'
+
+    _TESTS = [{
+        # QRR (文化放送) station provides <desc>
+        'url': 'https://radiko.jp/#!/ts/QRR/20210425101300',
+        'only_matching': True,
+    }, {
+        # FMT (TOKYO FM) station does not provide <desc>
+        'url': 'https://radiko.jp/#!/ts/FMT/20210810150000',
+        'only_matching': True,
+    }, {
+        'url': 'https://radiko.jp/#!/ts/JOAK-FM/20210509090000',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        station, video_id = self._match_valid_url(url).groups()
+        vid_int = unified_timestamp(video_id, False)
+
+        auth_token, area_id = self._auth_client()
+
+        prog, station_program, ft, ft_str, to_str = self._find_program(video_id, station, vid_int)
+
+        title = prog.find('title').text
+        description = clean_html(prog.find('info').text)
+        station_name = station_program.find('.//name').text
+
+        formats = self._extract_formats(video_id, station, False, ft, vid_int, auth_token, area_id, {
+            'start_at': ft_str,  # begin time of the radio
+            'ft': ft_str,  # same as start_id
+            'end_at': to_str,  # end time of the radio
+            'to': to_str,  # same as end_at
+            'seek': video_id,
+        })
+
+        return {
+            'id': video_id,
+            'title': title,
+            'description': description,
+            'uploader': station_name,
+            'uploader_id': station,
+            'timestamp': vid_int,
+            'formats': formats,
+            # we have to mark this live since they behave as if it's a live
```

Comment:
wdym? it IS live unless I am mistaken. Note that live doesn't mean the video is being recorded and played back at real time; just that it is a livestream (eg: youtube premiers are considered live)

Reply:
okay